chainer/variable.py

```diff
@@ -1614,7 +1614,8 @@ def add_cand(cand):
             i for i, x in enumerate(inputs) if x.requires_grad
         ])
         outputs = [y() for y in func.outputs]  # access via weak ref
-        out_grad = tuple([grads.pop(y) for y in outputs])
+        out_grad = tuple([grads.pop(y) if y.creator_node else None
```

Comment:
- `y` can be `None` if the output (node) is deleted.
- `func is None` is better because `func.__bool__` might be overridden.
```suggestion
        out_grad = tuple([grads.pop(y)
                          if y is not None and y.creator_node is not None
                          else None
```